[Doc] Update external.md (#6424)
taichi-dev/taichi: docs/lang/articles/basic/external.md

copy(x, y.clone()) # correct
copy(x, y.contiguous()) # correct
```

> ## Can I use `@ti.kernel` to accelerate a NumPy function?

Reviewer: ```suggestion
## FAQ

### Can I use `@ti.kernel` to accelerate a NumPy function?
```

> Users frequently ask whether they can call NumPy functions to process an array inside the Taichi scope. For example, like the follows:

Reviewer: ```suggestion
You may try to call NumPy functions to process an array inside the Taichi scope, as the following example shows:
```

```python
import numpy as np

@ti.kernel
def invalid_sum(arr: ti.types.ndarray()) -> float:
    return np.sum(arr)
```

> This is *not allowed*: Taichi doesn't compile NumPy's functions. This is a major difference between Taichi and other Python accelerating frameworks like Numba.

Reviewer: ```suggestion
This is *not allowed*. Unlike other Python acceleration frameworks, such as Numba, Taichi does not compile NumPy functions. 
```

Reviewer 2: *not allowed* -> *not supported* sounds more friendly.

> If you want to use a Numpy function which Taichi does not offer its counterpart functionality, you can call it in the Python scope as usual, and pass the result array  to Taichi kernels via `ti.types.ndarray()`.

Reviewer: ```suggestion
If you want to use a NumPy function to complement Taichi, you can call the function in the Python scope as usual and pass the generated array to Taichi kernels via `ti.types.ndarray()`.
```

Reviewer 2: @neozhaoliang could you provide a code example here?

copy(x, y.clone()) # correct
copy(x, y.contiguous()) # correct
```

## FAQ

### Can I use `@ti.kernel` to accelerate a NumPy function?

> You may try to call NumPy functions to process an array inside the Taichi scope, as the following example shows:

Reviewer 3: ```suggestion
Unlike other Python acceleration frameworks, such as Numba, Taichi does not compile NumPy functions. Calling NumPy functions inside the Taichi scope is not supported, as the following example shows:
```

```python
import numpy as np

@ti.kernel
def invalid_sum(arr: ti.types.ndarray()):
    total = np.sum(arr)

>     ...

Reviewer 3: We should put the error message in the comment so that it's obvious non-supported case.

```

> This is *not supported*: Taichi doesn't compile NumPy's functions. This is a major difference between Taichi and other Python accelerating frameworks like Numba.

Reviewer: ```suggestion
This is *not supported*. Unlike other Python acceleration frameworks, such as Numba, Taichi does not compile NumPy functions. 
```

> If you want to use a Numpy function which Taichi does not offer its counterpart functionality, you can call it in the Python scope as usual, and pass the result array  to Taichi kernels via `ti.types.ndarray()`. For example:

Reviewer: ```suggestion
If you want to use a NumPy function, which lacks a counterpart in Taichi, you can call the function in the Python scope as usual and pass the processed array to Taichi kernels via `ti.types.ndarray()`. For example:
```
